### PR TITLE
Improve GCC calls

### PR DESCRIPTION
**Pull request by jspricke**

### CODE REVIEW

**Comment by Cielquan** [comment]
Could you please update the PR to the new state of the repo. The file rstcheck.py no longer exists. It is now rstcheck/init.py.

And could you please explain to me what those changes intend to solve. I understand the python code but not the intention behind. And why removing the `-std` option?

**Comment by jspricke** [comment]
>Could you please update the PR to the new state of the repo. The file rstcheck.py no longer exists. It is now rstcheck/init.py.

Done.

>And could you please explain to me what those changes intend to solve. I understand the python code but not the intention behind.

C/C++ projects often need additional compiler flags for compilation. 
These could for example be extra include directories (-I) or additional 
warnings (-Wall). The standard way to adopt them is by setting 
CFLAGS/CXXFLAGS in your process environment.

>And why removing the `-std` option?

 From man gcc:

| When a base standard is specified, the compiler accepts all programs following that standard plus those using GNU extensions that do not contradict it.

In other words, it rejects programs that need newer standards.

The default standard in gcc 11 are gnu17 (GNU dialect of ISO C17) and 
gnu++17 (GNU dialect of 2017 ISO C++), so by limiting the standards to 
20 year old versions, rstcheck rejects a lot of valid programs. You 
could set -std in CFLAGS/CXXFLAGS if needed.

**Comment by Cielquan** [comment]
LGTM.
Thanks for the explanation. Appreciate it.
